src/main/java/com/redhat/cloud/custompolicies/app/rest/PolicyCrudService.java

```diff
@@ -226,14 +234,49 @@ public Response deletePolicy(@PathParam("id") Long policyId) {
       builder = Response.status(Response.Status.BAD_REQUEST);
     } else {
       policy.delete(policy);
+      engine.deleteTrigger(policy.triggerId, user.getAccount());
       builder = Response.ok(policy);
     }
 
     return builder.build();
 
+  }
+
+  @Operation(summary = "Update a single policy for a customer by its id")
+  @PUT
+  @Path("/{policyId}")
+  @APIResponse(responseCode = "200", description = "Policy updated")
+  @APIResponse(responseCode = "400", description = "Update failed")
+  @Transactional
+  public Response updatePolicy(@PathParam("policyId") Long policyId, @Valid Policy policy) {
+
+    System.out.println("update, passed in is " + policy);
+    Policy tmp = Policy.findById(user.getAccount(), policyId);
+    System.out.println("Policy found " + tmp);
+
+    ResponseBuilder builder ;
+    if (tmp==null) {
+      builder = Response.status(404, "Original policy not found");
+    } else {
+      if (!policy.id.equals(policyId)) {
+        builder = Response.status(400, "Invalid policy");
+        return builder.build();
+      }
+      tmp.populateFrom(policy);
+      System.out.println("Persisting now " + tmp);
+      tmp.persist(tmp);
+      System.out.println("update: new policy persisted");
+      FullTrigger trigger = new FullTrigger(policy);
+      trigger.trigger.id = tmp.triggerId;
+      engine.update(tmp.triggerId, trigger, user.getAccount());
```

Comment:
@burmanm This sends a `PUT /hawkular/alerts/triggers/6a9da736-6737-450d-b96c-466fdf3fe101 HTTP/1.1  (application/json)`

With that content (as json, this is the wireshark rep)
```
JavaScript Object Notation: application/json
    Object
        Member Key: conditions
            Array
                Object
                    Member Key: dataId
                        String value: insights_report
                        Key: dataId
                    Member Key: expression
                        String value: cores = 3
                        Key: expression
                    Member Key: triggerMode
                        String value: FIRING
                        Key: triggerMode
                    Member Key: type
                        String value: EVENT
                        Key: type
            Key: conditions
        Member Key: trigger
            Object
                Member Key: enabled
                    False value
                    Key: enabled
                Member Key: id
                    String value: 6a9da736-6737-450d-b96c-466fdf3fe101
                    Key: id
                Member Key: name
                    String value: test2
                    Key: name
            Key: trigger
```

This is the same format I use to initially send FullTriggers via POST, which succeeds. This now ends up in an error
 ```
"java.lang.IllegalStateException: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "conditions" (class org.hawkular.alerts.api.model.trigger.Trigger), not marked as ignorable (22 known properties: "enabled", "tenantId", "eventText", "autoResolveMatch", "name", "context", "memberOf", "autoEnable", "eventType", "dataIdMap", "eventCategory", "tags", "type", "id", "description", "severity", "firingMatch", "autoResolveAlerts", "autoDisable", "source", "actions", "autoResolve"])
 at [Source: (String)"{"conditions":[{"dataId":"insights_report","expression":"cores = 3","triggerMode":"FIRING","type":"EVENT"}],"trigger":{"enabled":false,"id":"6a9da736-6737-450d-b96c-466fdf3fe101","name":"test2"}}"; line: 1, column: 16] (through reference chain: org.hawkular.alerts.api.model.trigger.Trigger["conditions"])"

```

Do you have any idea?

Reply:
The URL is incorrect. It's ``/hawkular/alerts/triggers/trigger/{triggerId}`` for FullTrigger update.